src/useRouteQueryParam/formats/RouteParam.spec.ts

```diff
@@ -0,0 +1,11 @@
+import { describe, expect, it } from 'vitest'
+import { RouteParam, isRouteParamClass } from '@/useRouteQueryParam/formats/RouteParam'
+
+describe('RouteParam', () => {
+
+  it('type guard is valid', () => {
+    expect(isRouteParamClass(RouteParam)).toBe(true)
```

Comment:
if this test were to fail, wouldn't the build itself fail?

Reply:
No, this is essentially testing the logic inside the `isRouteParamClass` function. That could be totally jacked up and the build would still pass as long as the logic returned a boolean.